pdns/auth-packetcache.cc

```diff
@@ -162,7 +163,15 @@ void AuthPacketCache::insert(DNSPacket& q, DNSPacket& r, unsigned int maxTTL)
 
     /* no existing entry found to refresh */
     mc.d_map.insert(entry);
-    (*d_statnumentries)++;
+
+    if (*d_statnumentries >= d_maxEntries) {
+      /* remove the least recently inserted or replaced entry */
+      auto& sidx = mc.d_map.get<SequencedTag>();
```

Comment:
Question unrelated to this change: why do we actively manage `d_statnumentries`? Does this map not expose a cheap `.size()` ?

Reply:
We have several shards, so we would need to iterate over all of them, taking a lock in the process, to know the total number of entries.

Reply:
Ah of course, thanks.